cli_tools/osconfig_agent/main.go

```diff
@@ -42,22 +101,42 @@ func strIn(s string, ss []string) bool {
 	return false
 }
 
-func main() {
-	flag.Parse()
-	patchInit()
-
-	ctx := context.Background()
-
+func run(ctx context.Context) {
 	client, err := osconfig.NewClient(ctx, option.WithEndpoint(*endpoint), option.WithCredentialsFile(*oauth))
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("NewClient Error:", err)
 	}
 
-	res, err := lookupConfigs(ctx, client, *resource)
+	res, err := getResource(*resource)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("getResource error:", err)
 	}
-	patchManager(res.PatchPolicies)
 
-	//runUpdates()
+	patchInit()
+	ticker := time.NewTicker(interval)
+	for {
+		res, err := lookupConfigs(ctx, client, res)
+		if err != nil {
+			log.Println("ERROR:", err)
+		} else {
+			runPackageConfig(res)
```

Comment:
`PackageConfig` isn't really the right name here is it? This call will be setting registry settings as well. And maybe more in the future. How about something like `setOsConfig()`? 

Also, we might want to tweak our api so this would be something like `setOsConfig(res.osConfig)`. But I don't think we need to worry about that right now.

Reply:
Ack, changed to setOsConfig. And yeah I think that api change would make sense.